.build/docker/_docker_run.sh

```diff
@@ -111,22 +112,34 @@ container_name="cassandra_${dockerfile/.docker/}_${un_script_name}_jdk${java_ver
 #  change ant's build directory to $DIST_DIR
 #  set java to java_version
 #  execute the run_script
+# docker_command="export ANT_OPTS=\"-Dbuild.dir=\${DIST_DIR} ${CASSANDRA_DOCKER_ANT_OPTS}\" ; \
+#                 source \${CASSANDRA_DIR}/.build/docker/_set_java.sh ${java_version} ; \
+#                 \${CASSANDRA_DIR}/.build/${run_script} ${@:4} ; exit \$? "
+
 docker_command="export ANT_OPTS=\"-Dbuild.dir=\${DIST_DIR} ${CASSANDRA_DOCKER_ANT_OPTS}\" ; \
-                source \${CASSANDRA_DIR}/.build/docker/_set_java.sh ${java_version} ; \
-                \${CASSANDRA_DIR}/.build/${run_script} ${@:4} ; exit \$? "
+                source /home/build/cassandra/cassandra/.build/docker/_set_java.sh ${java_version} ; \
+                /home/build/cassandra/.build/${run_script} ${@:4} ; exit \$? "
 
 # run without the default seccomp profile
 # re-use the host's maven repository
+# container_id=$(docker run --name ${container_name} -d --security-opt seccomp=unconfined --rm \
+#     -v "${cassandra_dir}":/home/build/cassandra -v ~/.m2/repository/:/home/build/.m2/repository/ -v "${build_dir}":/dist \
+#     ${docker_volume_opt} \
+#     ${image_name} sleep 1h)
 container_id=$(docker run --name ${container_name} -d --security-opt seccomp=unconfined --rm \
     -v "${cassandra_dir}":/home/build/cassandra -v ~/.m2/repository/:/home/build/.m2/repository/ -v "${build_dir}":/dist \
     ${docker_volume_opt} \
     ${image_name} sleep 1h)
 
-echo "Running container ${container_name} ${container_id}"
 
-docker exec --user root ${container_name} bash -c "\${CASSANDRA_DIR}/.build/docker/_create_user.sh build $(id -u) $(id -g)"
+echo "Running container ${container_name} ${container_id}"
+#sleep 3600
+#docker exec --user root ${container_name} bash -c "\${CASSANDRA_DIR}/.build/docker/_create_user.sh build $(id -u) $(id -g)"
+docker exec --user root ${container_name} bash -c "/home/build/cassandra/.build/docker/_create_user.sh build $(id -u) $(id -g)"
 docker exec --user build ${container_name} bash -c "${docker_command}"
 RETURN=$?
+current_build_dir=$(basename "${build_dir}")
+docker cp ${container_name}:"/home/build/cassandra/build/${current_build_dir}" /home/jenkins/agent/workspace/k8s-e2e/build
```

Comment:
everything under `/dist` should already be available at `${build_dir}`  on the docker host
because of the mount ` -v "${build_dir}":/dist` on this line https://github.com/thelastpickle/cassandra/pull/2/files/04dde10476e61bd9d399e0fdbbf60a1aa1b2177b#diff-e8f0cb5844d900a3d89a43743814930e66575abc2b6731c07d6a691de51a4d65R129-R130

(`/home/build/cassandra/build/${current_build_dir}` should not be the build directory inside the docker container.)

Remember the newline character breaking the mount from working.

Also, inside the docker container we need to make sure that both `${DIST_DIR}` and `export ANT_OPTS=\"-Dbuild.dir=\${DIST_DIR}` is set.

The latter happens here: https://github.com/thelastpickle/cassandra/pull/2/files/04dde10476e61bd9d399e0fdbbf60a1aa1b2177b#diff-e8f0cb5844d900a3d89a43743814930e66575abc2b6731c07d6a691de51a4d65R115

I cannot see where we are setting `DIST_DIR=/dist` inside the container.

Reply:
as Nishant and Richa explained today, the problem so far here has been the files can be found in three places.

 - k8s node (which is also the docker host)
 - k8s pod (which is where the jenkins agent, and the jenkins groovy script runs on)
 - docker container (which mounts from the k8s node)

The jenkins script doesn't have access to the build_dir (only the k8s node and docker container has access).

Nishant and Richa are working on how to provide/share/mount the cassandra_dir (which includes all build_dir) from k8s nodes to pods.